packages/radio-player/src/radio-player.ts

```diff
@@ -0,0 +1,738 @@
+        @forward-button-pressed=${this.forwardButtonHandler}
+        @volumeChange=${this.volumeChanged}
+        @playbackRateChange=${this.changePlaybackRate}
+        @next-section-button-pressed=${this.nextSectionButtonHandler}
+        @prev-section-button-pressed=${this.prevSectionButtonHandler}
+      >
+      </playback-controls>
+    `;
+  }
+
+  private get scrubberBarMarkerPercentages(): number[] {
+    const percentages: number[] = [0];
+    this.zonesOfSilence.forEach(zone => {
+      percentages.push(zone.startPercent);
+      percentages.push(zone.endPercent);
+    });
+    percentages.push(100);
+    return percentages;
+  }
+
+  private get scrubberBarTemplate(): TemplateResult {
+    return html`
+      <scrubber-bar
+        .sectionMarkerPercentages=${this.scrubberBarMarkerPercentages}
+        .value=${this.percentComplete}
+        @valuechange=${this.valueChangedFromScrub}
+      >
+      </scrubber-bar>
+    `;
+  }
+
+  private get transcriptViewTemplate(): TemplateResult {
+    return html`
+      <div class="transcript-container">
+        <transcript-view
+          .config=${this.transcriptConfig}
+          .currentTime=${this.currentTime}
+          @transcriptEntrySelected=${this.transcriptEntrySelected}
+        >
+        </transcript-view>
+      </div>
+    `;
+  }
+
+  private get searchSectionTemplate(): TemplateResult {
+    // The mobile and desktop search sections work similarly, but the mobile version has
+    // a dropdown area on it and the desktop version does not.
+    // This is a case where the functionality is different enough to have two instances
+    // of it instead of one and just show and hide them based on the media query.
+    return html`
+      <div class="search-section">
+        <expandable-search-bar
+          searchTerm=${this.searchTerm}
+          .quickSearches=${this.quickSearches}
+          @inputchange=${this.updateSearchTerm}
+          @enterKeyPressed=${this.searchEnterKeyPressed}
+          @searchCleared=${this.searchCleared}
+        >
+        </expandable-search-bar>
+        <div class="search-results-info">
+          ${this.searchResultsSwitcherTemplate} ${this.noSearchResultsTemplate}
+        </div>
+      </div>
+    `;
+  }
+
+  private get searchResultsSwitcherTemplate(): TemplateResult {
+    return html`
+      <search-results-switcher
+        class="${this.shouldShowSearchResultSwitcher ? '' : 'hidden'}"
+        @searchResultIndexChanged=${this.searchResultIndexChanged}
+      >
+      </search-results-switcher>
+    `;
+  }
+
+  private get noSearchResultsTemplate(): TemplateResult {
+    return html`
+      <div
+        class="no-search-results-message ${this.shouldShowNoSearchResultMessage ? '' : 'hidden'}"
+      >
+        No search results.
+      </div>
+    `;
+  }
+
+  private get quickSearches(): QuickSearchEntry[] {
+    if (!this.config) {
+      return [];
+    }
+
+    const entries: QuickSearchEntry[] = this.config.quickSearches.map(
```

Comment:
q: is it given that `quickSearches` is always within `this.config`?

Reply:
Correct, the radio player config contains quick searches as an array of strings: 
```
config = {
  url: "foo.mp3",
  title: "Bar Channel 3",
  quickSearches: ["foo", "bar", "baz"],
  ...
}
```

This takes those values and converts them into `QuickSearchEntry` objects for use the in the quick search picker.